package.json

```diff
@@ -555,6 +555,77 @@
 					"type": "integer",
 					"default": 500,
 					"description": "Set a limit for number of edges displayed in a graph visualization"
+				},
+				"cosmosDB.graph.viewSettings": {
+					"type": "object",
+					"description": "Settings for graph visualization. Currently only the 'default' group is supported.",
```

Comment:
I don't understand the purpose of the 'default' group. Can you explain what other 'groups' you plan to support in the future? Just an example or two?

Reply:
What I want in the future is a way to have different settings for different graphs or sets of graphs.  My current thinking is that I would probably end up having named settings groups.  These could either be controlled by the user or auto-generated and not seen by the user in the UI. Then each group could specify which graphs it refers to.  This seems like the most flexible feature for the future.  I'd rather not be designing for the future with it now, but can't think of a way to do that that wouldn't require changes in the settings model later.  I'm open to suggestions, though.

The other main option I considered was like I'm doing with the vertex settings groups - having an array of objects, and the object contains some properties to filter them.  But that seems even more confusing.  Thoughts?

Reply:
Yes - I think it makes more sense to have an array and mirror the structure of the vertex settings groups. What about this structure:
```json
"cosmosDB.graphViewSettings": [
    {
        "appliesTo": [
            {
                "accountName": "",
                "databaseName": "",
                "graphName": ""
            }
        ],
        "vertexViewSettings": [
            {
                "appliesTo": [
                    {
                        "vertexLabel": ""
                    }
                ],
                "displayProperty": [],
                "color": ""
            }
        ]
    }
]
```

You'll notice the name "graphViewSettings" mirrors "vertexViewSettings" and they both have the same "appliesTo" array (I also considered naming this "filter" instead of "appliesTo")

Reply:
And for now, you could just not support "appliesTo" at the graph level (It would be the same as only having "default" as an option)

Reply:
Okay, I like that.